src/solve/SimulationParameters.h

```diff
@@ -70,61 +74,139 @@ struct SimulationParameters {
+ * @param connections Vector storing the connections between blocks
+ * @param config The JSON configuration containing all the vessels
+ * @param vessel_id_map Map between vessel names and IDs
+ */
 void create_vessels(
     Model& model,
     std::vector<std::tuple<std::string, std::string>>& connections,
     const nlohmann::json& config, std::map<int, std::string>& vessel_id_map);
 
+/**
+ * @brief Handle the creation of external coupling blocks and connections with
+ * other blocks
+ *
+ * @param model The model the block is associated with
+ * @param connections Vector storing the connections between blocks
+ * @param config The JSON configuration containing all the external coupling
+ * blocks
+ * @param vessel_id_map Map between vessel names and IDs
+ * @param bc_type_map Map between boundary condition names and their types
+ */
 void create_external_coupling(
     Model& model,
     std::vector<std::tuple<std::string, std::string>>& connections,
     const nlohmann::json& config, std::map<int, std::string>& vessel_id_map,
     std::map<std::string, std::string>& bc_type_map);
 
+/**
+ * @brief Handle the creation of boundary condition blocks
+ *
+ * @param model The model the block is associated with
+ * @param config The JSON configuration containing all the boundary condition
+ * blocks
+ * @param bc_type_map Map between boundary condition names and their types
+ * @param closed_loop_bcs List of boundary conditions that should be connected
+ * to a closed loop heart block
+ */
 void create_boundary_conditions(Model& model, const nlohmann::json& config,
                                 std::map<std::string, std::string>& bc_type_map,
                                 std::vector<std::string>& closed_loop_bcs);
 
+/**
+ * @brief Handle the creation of junctions and their connections
+ *
+ * @param model The model the block is associated with
+ * @param connections Vector storing the connections between blocks
+ * @param config The JSON configuration containing all the junctions
+ * @param vessel_id_map Map between vessel names and IDs
+ */
 void create_junctions(
     Model& model,
     std::vector<std::tuple<std::string, std::string>>& connections,
     const nlohmann::json& config, std::map<int, std::string>& vessel_id_map);
 
+/**
+ * @brief Handle the creation of closed-loop blocks and associated connections
+ *
+ * @param model The model the block is associated with
+ * @param connections Vector storing the connections between blocks
+ * @param config The JSON configuration containing all the closed loop blocks
+ * @param closed_loop_bcs List of boundary conditions that should be connected
+ * to a closed loop heart block
+ */
 void create_closed_loop(
     Model& model,
     std::vector<std::tuple<std::string, std::string>>& connections,
     const nlohmann::json& config, std::vector<std::string>& closed_loop_bcs);
 
+/**
+ * @brief Handle the creation of valves and their associated connections
```

Comment:
Is there a way to require these in Doxygen?

Reply:
This is already required if the file is seen by Doxygen. The loophole is when the file isn't seen by Doxygen because of a missing `@file` tag on top (comment below).